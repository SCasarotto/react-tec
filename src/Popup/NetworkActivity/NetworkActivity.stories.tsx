import React from 'react';

import { Story, Meta } from '@storybook/react';

import { NetworkActivity, NetworkActivityProps } from './NetworkActivity';

export default {
  title: 'Popup/NetworkActivity',
  component: NetworkActivity,
  argTypes: {},
} as Meta;

const Template: Story<NetworkActivityProps> = (args) => (
  <NetworkActivity {...args} />
);

export const Default = Template.bind({});
Default.args = {
  message: 'Loading...',
  visible: true,
};
