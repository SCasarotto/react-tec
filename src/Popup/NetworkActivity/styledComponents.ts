import styled from 'styled-components';

import { Spinner as S } from '../../Layout/Spinner';
import { Popup as P } from '../Popup';

export const Popup = styled(P)`
  display: flex;
  justify-content: center;
  align-items: center;

  .PopupContent {
    background-color: rgba(0, 0, 0, 0.5);
    padding: 60px;
    text-align: center;
  }
`;
export const Spinner = styled(S)`
  margin-bottom: 30px;
`;
export const Message = styled.p`
  position: relative;
  width: 100%;
  text-align: center;
  font-size: 24px;
  margin: 0px;
  color: ${(props) => props.theme.white};
`;
