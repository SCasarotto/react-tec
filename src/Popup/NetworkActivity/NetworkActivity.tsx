import React from 'react';

import { PopupProps } from '../Popup';
import { Popup, Spinner, Message } from './styledComponents';

export interface NetworkActivityProps extends PopupProps {
  message?: string;
}
/**
 *
 * NetworkActivity is an overlay to indicate some network request or long running process is taking place.
 *
 */
export const NetworkActivity: React.FC<NetworkActivityProps> = (props) => {
  const { visible, className = '', message } = props;

  return (
    <Popup visible={visible} className={`tec-networkActivity ${className}`}>
      <Spinner size='large' className='tec-networkActivitySpinner' />
      {message && (
        <Message className='tec-networkActivityMessage'>{message}</Message>
      )}
    </Popup>
  );
};
