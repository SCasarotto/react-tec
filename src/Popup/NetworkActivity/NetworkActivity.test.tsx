import React from 'react';

import { render } from '@testing-library/react';

import { NetworkActivity } from './NetworkActivity';

describe('NetworkActivity', () => {
  // Renders Without Error
  it('matches snapshot', () => {
    const { asFragment } = render(<NetworkActivity visible />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Class Name
  it('accepts custom className', () => {
    const { container } = render(
      <NetworkActivity visible className='myClassName' />,
    );
    expect(container.firstChild).toHaveClass('myClassName');
  });
});
