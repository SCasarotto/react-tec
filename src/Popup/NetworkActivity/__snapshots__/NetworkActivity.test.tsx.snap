// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkActivity matches snapshot 1`] = `
<DocumentFragment>
  .c3 {
  text-align: center;
}

.c5 {
  position: relative;
  display: inline-block;
  width: 80px;
  height: 80px;
}

.c6 {
  position: absolute;
  border-radius: 50%;
  overflow: hidden;
  border-top: 3px solid transparent;
  border-right: 1px solid transparent;
  border-bottom: 1px solid transparent;
  border-left: 1px solid transparent;
  left: 0%;
  top: 0%;
  width: 100%;
  height: 100%;
  border-top-color: #000000;
  -webkit-animation: TESpinnerLoading 1s linear infinite;
  animation: TESpinnerLoading 1s linear infinite;
}

.c7 {
  position: absolute;
  border-radius: 50%;
  overflow: hidden;
  border-top: 3px solid transparent;
  border-right: 1px solid transparent;
  border-bottom: 1px solid transparent;
  border-left: 1px solid transparent;
  left: 8.333333333333332%;
  top: 8.333333333333332%;
  width: 83.33333333333334%;
  height: 83.33333333333334%;
  border-top-color: #333333;
  -webkit-animation: TESpinnerLoading 1.125s linear infinite;
  animation: TESpinnerLoading 1.125s linear infinite;
}

.c8 {
  position: absolute;
  border-radius: 50%;
  overflow: hidden;
  border-top: 3px solid transparent;
  border-right: 1px solid transparent;
  border-bottom: 1px solid transparent;
  border-left: 1px solid transparent;
  left: 16.666666666666664%;
  top: 16.666666666666664%;
  width: 66.66666666666667%;
  height: 66.66666666666667%;
  border-top-color: #666666;
  -webkit-animation: TESpinnerLoading 1.2857142857142856s linear infinite;
  animation: TESpinnerLoading 1.2857142857142856s linear infinite;
}

.c9 {
  position: absolute;
  border-radius: 50%;
  overflow: hidden;
  border-top: 3px solid transparent;
  border-right: 1px solid transparent;
  border-bottom: 1px solid transparent;
  border-left: 1px solid transparent;
  left: 25%;
  top: 25%;
  width: 50%;
  height: 50%;
  border-top-color: #9a9a9a;
  -webkit-animation: TESpinnerLoading 1.5s linear infinite;
  animation: TESpinnerLoading 1.5s linear infinite;
}

.c10 {
  position: absolute;
  border-radius: 50%;
  overflow: hidden;
  border-top: 3px solid transparent;
  border-right: 1px solid transparent;
  border-bottom: 1px solid transparent;
  border-left: 1px solid transparent;
  left: 33.33333333333333%;
  top: 33.33333333333333%;
  width: 33.333333333333336%;
  height: 33.333333333333336%;
  border-top-color: #cccccc;
  -webkit-animation: TESpinnerLoading 1.7999999999999998s linear infinite;
  animation: TESpinnerLoading 1.7999999999999998s linear infinite;
}

.c11 {
  position: absolute;
  border-radius: 50%;
  overflow: hidden;
  border-top: 3px solid transparent;
  border-right: 1px solid transparent;
  border-bottom: 1px solid transparent;
  border-left: 1px solid transparent;
  left: 41.666666666666664%;
  top: 41.666666666666664%;
  width: 16.666666666666675%;
  height: 16.666666666666675%;
  border-top-color: #ffffff;
  -webkit-animation: TESpinnerLoading 2.2499999999999996s linear infinite;
  animation: TESpinnerLoading 2.2499999999999996s linear infinite;
}

.c0 {
  position: fixed;
  width: 100%;
  height: 100%;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  background-color: rgba(0,0,0,0.5);
  z-index: 1000;
  overflow-y: auto;
  -webkit-transition: visibility 0.25s ease-in-out,opacity 0.25s ease-in-out;
  transition: visibility 0.25s ease-in-out,opacity 0.25s ease-in-out;
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: relative;
  border-radius: 20px;
  margin: 50px auto;
  max-width: 650px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .PopupContent {
  background-color: rgba(0,0,0,0.5);
  padding: 60px;
  text-align: center;
}

.c4 {
  margin-bottom: 30px;
}

<div
    class="c0 tec-popup c1 tec-networkActivity "
  >
    <div
      class="c2 tec-popupContent"
    >
      <div
        class="c3 tec-spinner c4 tec-networkActivitySpinner"
      >
        <div
          class="c5 tec-spinnerContainer"
        >
          <div
            class="c6 tec-spinnerRing tec-spinnerRing1"
          />
          <div
            class="c7 tec-spinnerRing tec-spinnerRing2"
          />
          <div
            class="c8 tec-spinnerRing tec-spinnerRing3"
          />
          <div
            class="c9 tec-spinnerRing tec-spinnerRing4"
          />
          <div
            class="c10 tec-spinnerRing tec-spinnerRing5"
          />
          <div
            class="c11 tec-spinnerRing tec-spinnerRing6"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
